docs/programming_guide/client_controlled_workflows.rst

```diff
@@ -0,0 +1,852 @@
+.. _client_controlled_workflows:
+
+##########################################################
+Client Controlled Workflows for Peer-to-Peer Communication
+##########################################################
+
+Before version 2.4, all federating learning workflows (fed-average, cyclic controller, cross-site evaluation) were server controlled,
+implemented with the server-side controller API. In these workflows, FL clients get tasks assigned by the controller, execute the tasks,
+and submit results back to the server. It is up to the server when and how to assign tasks to clients.
```

Comment:
Good explanation, a question regarding the doc structure,
Should we separate the texts inside this file to two parts:
-  "user_guide": explains the motivation AND how to configure it via json files, more user-facing
-  "programming_guide": explains how to write it / customize it, more technical

Reply:
I think for now, maybe keep it as is. The organization is now grouped at a sub-level with other controllers too